### PR TITLE
Global colors fix

### LINKED ISSUE #1192: Block error when there is a null color in Global settings
The console shows the following error:
Screen Shot 2021-05-18 at 1 41 00 PM
Screen Shot 2021-05-18 at 1 41 14 PM

Should be able to handle the `null` values in global colors

### PR DESCRIPTION
**Pull request by blankart**
Fixes #1192

### CODE REVIEW

File: src/plugins/global-settings/colors/color-picker.js [hunk 1]
```diff
@@ -226,7 +226,17 @@ const ColorPickers = props => {
 	const colors = Array.isArray( _colors ) ? _colors : []
 
 	// Enable reset if there are Stackable global colors.
-	const disableReset = useMemo( () => ! colors.some( color => color.slug && color.slug.includes( 'stk-global-color' ) ), [ JSON.stringify( colors ) ] )
+	const disableReset = useMemo( () => ! colors.some( color => {
+		if ( typeof color !== 'object' ) {
+			return false
+		}
+
+		if ( typeof color.slug === 'string' && color.slug.match( /stk-global-color/ ) ) {
```
Comment (bfintal) [inline src/plugins/global-settings/colors/color-picker.js:234]: we used `color.slug.includes( 'stk-global-color' )` before, please use that too

File: src/plugins/global-settings/colors/color-picker.js [hunk 2]
```diff
@@ -336,12 +346,16 @@ const ColorPickers = props => {
...
+					return null
+				}
+
 				return (
 					<ColorOption
 						key={ index }
 						color={ color.color }
 						name={ color.name }
-						locked={ ! color.slug.match( /^stk-/ ) }
+						locked={ ! ( color.slug || '' ).match( /^stk-/ ) }
```
Comment (bfintal) [inline src/plugins/global-settings/colors/color-picker.js:358]:
```suggestion
						locked={ ! ( color.slug || '' ).startsWith( 'stk-' ) }
```

File: src/plugins/global-settings/colors/editor-loader.js [hunk 3]
```diff
@@ -13,33 +13,42 @@ import domReady from '@wordpress/dom-ready'
...
 
 const renderGlobalStyles = ( newColors, setStyles ) => {
 	// Output all our --stk-global-colors.
 	const styleRules = newColors.map( color => {
-		if ( color.slug.match( /^stk-global-color/ ) ) {
+		if ( typeof color !== 'object' ) {
+			return null
+		}
+
+		if ( typeof color.slug === 'string' && color.slug.match( /^stk-global-color/ ) ) {
```
Comment (bfintal) [inline src/plugins/global-settings/colors/editor-loader.js:25]:
```suggestion
		if ( typeof color.slug === 'string' && color.slug.startsWith( 'stk-global-color' ) ) {
```

File: src/plugins/global-settings/colors/editor-loader.js [hunk 4]
```diff
@@ -13,33 +13,42 @@ import domReady from '@wordpress/dom-ready'
...
+	setStyles( `:root { ${ compact( styleRules ).join( '' ) }}` )
 
 	// Output all the rgba colors, detect the actual color values.
 	const rgbaStyleRules = newColors.map( color => {
-		if ( color.slug.match( /^stk-global-color/ ) ) {
+		if ( typeof color !== 'object' ) {
+			return null
+		}
+
+		if ( typeof color.slug === 'string' && color.slug.match( /^stk-global-color/ ) ) {
```
Comment (bfintal) [inline src/plugins/global-settings/colors/editor-loader.js:40]:
```suggestion
		if ( typeof color.slug === 'string' && color.slug.startsWith( 'stk-global-color' ) ) {
```

File: src/plugins/global-settings/colors/store.js [hunk 5]
```diff
@@ -60,6 +60,10 @@ registerStore( 'stackable/global-colors', {
  * @return {Array} new stackable global color array
  */
 const convertBetaStackableColorsToRelease = colors => compact( ( colors || [] ).map( color => {
+	if ( typeof color !== 'object' ) {
+		return null
+	}
+
 	if ( color.fallback && color.colorVar ) {
 		// Let us not include global colors linked to theme colors.
 		return color.slug.match( /^stk-global-color/ ) ?
```
Comment (bfintal) [inline src/plugins/global-settings/colors/store.js:69]:
```suggestion
		return color.slug.startsWith( 'stk-global-color' ) ?
```

File: src/plugins/global-settings/colors/color-picker.js (hunk 1 above)
Comment (blankart) [inline src/plugins/global-settings/colors/color-picker.js:234]:
```suggestion
		if ( typeof color.slug === 'string' && color.slug.includes( 'stk-global-color' ) ) {
```

**Pull request merged**